File: go/p003.go
package main

import "fmt"

func main() {
	var (
		n    int64 = 600851475143
		k    int64 = 2
		last int64 = k
	)
	for n > 1 {
		if n%k == 0 {
			last = k
			for n%k == 0 {
				n /= k
			}
		}
		k += 1
	}
	fmt.Println(last)
	// output: 6
}

File: rust/p003.rs
fn main() {
	let mut	n: i64 = 600851475143;
	let mut	k: i64 = 2;
	let mut	last: i64 = k;
	while n > 1 {
		if n%k == 0 {
			last = k;
			while n%k == 0 {
				n /= k;
			}
		}
		k += 1;
	}
	println(last.to_str())
	// output: 6857
}

